Detectors/EMCAL/workflow/src/CellRecalibratorSpec.cxx

```diff
@@ -86,45 +83,29 @@ void CellRecalibratorSpec::finaliseCCDB(o2::framework::ConcreteDataMatcher& matc
   }
 }
 
-std::optional<o2::emcal::Cell> CellRecalibratorSpec::getCalibratedCell(const o2::emcal::Cell& input) const
-{
-  if (isRunBadChannlCalibration() && mBadChannelMap) {
-    auto cellstatus = mBadChannelMap->getChannelStatus(input.getTower());
-    // reject bad and dead cells
-    if (cellstatus == BadChannelMap::MaskType_t::BAD_CELL || cellstatus == BadChannelMap::MaskType_t::DEAD_CELL) {
-      return std::optional<o2::emcal::Cell>();
-    }
-  }
-  auto celltime = input.getTimeStamp();
-  auto cellenergy = input.getEnergy();
-  if (isRunTimeCalibration() && mTimeCalibration) {
-    celltime -= mTimeCalibration->getTimeCalibParam(input.getTower(), input.getLowGain());
-  }
-  if (isRunGainCalibration() && mGainCalibration) {
-    cellenergy *= mGainCalibration->getGainCalibFactors(input.getTower());
-  }
-  o2::emcal::Cell outputcell(input.getTower(), cellenergy, celltime, input.getType());
-  return std::optional<o2::emcal::Cell>(outputcell);
-}
-
 void CellRecalibratorSpec::updateCalibObjects()
 {
-  auto badmapOld = mBadChannelMap;
-  auto timeCalibOld = mTimeCalibration;
+  bool hasUpdate = false;
   if (isRunBadChannlCalibration()) {
-    mBadChannelMap = mCalibrationHandler->getBadChannelMap();
-    if (badmapOld != mBadChannelMap) {
+    if (mCellRecalibrator.getBadChannelMap() != mCalibrationHandler->getBadChannelMap()) {
```

Comment:
not sure such an object is 100% reliable (unless the object never changes, for the 1st object this is surely fine): on update, the old cached object is destroyed and instead a new one is created, theoretically it may have the same pointer.

Reply:
I was also worrying about that. Anything I can use from the framework to identify whether the object has changed? Once the objects from the online calibration are used they often have a finer-than-run granularity.

Reply:
No, the framework just calls finaliseCCDB on the 1st `pc.inputs().get...` for updated object, the rest is on the author of the device.
To flag object update in the helper I use dedicated data members, e.g. https://github.com/AliceO2Group/AliceO2/blob/cb98b7e5f036024dfce73abf2b6bdb3ad432d1b4/GPU/TPCFastTransformation/CorrectionMapsHelper.h#L77-L83 which I then reser when consuming the object by https://github.com/AliceO2Group/AliceO2/blob/cb98b7e5f036024dfce73abf2b6bdb3ad432d1b4/GPU/TPCFastTransformation/CorrectionMapsHelper.h#L90

Reply:
Hi @shahor02 , I have changed the calib loader accordingly. Could you please review? The bitset size is chosen to be 16 in order to allow for other calibration objects to be handled by the calib loader and to make this a general tool.